glue/utils/matplotlib.py

```diff
@@ -176,3 +176,44 @@ def point_contour(x, y, data):
         return None
     xy = xy[0]
     return xy
+
+
+class AxesResizer(object):
+
+    def __init__(self, ax, margins):
+
+        self.ax = ax
+        self.margins = margins
+
+    def on_resize(self, event):
+
+        fig_width = self.ax.figure.get_figwidth()
+        fig_height = self.ax.figure.get_figheight()
+
+        x0 = self.margins[0] / fig_width
+        x1 = 1 - self.margins[1] / fig_width
+        y0 = self.margins[2] / fig_height
+        y1 = 1 - self.margins[3] / fig_height
+
+        dx = max(0.01, x1 - x0)
+        dy = max(0.01, y1 - y0)
+
+        self.ax.set_position([x0, y0, dx, dy]) 
+        self.ax.figure.canvas.draw()
+
+
+def fix_margins(axes, margins=[1, 1, 1, 1]):
```

Comment:
what do you think about the name `freeze_margins`? `fix` could be interpreted as implying that the default is `broken`.